### PR TITLE
chore: Add parent PR workflow to GH build actions

### PR DESCRIPTION
**Pull request by hnnasit**
*Description of changes:*
Create a parent job PR workflow action for accepting PRs similar to aws-sam-cli and aws-lambda-builders for easier maintenance.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by mndeveci — APPROVED** [review]

**Review by mildaniel — APPROVED** [review]

**Comment by mildaniel** [comment]
After merging, we'll need to update the branch settings to make this the approval workflow instead of the individual configured workflows today.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -4,6 +4,22 @@ on:
   pull_request
 
 jobs:
+  run-workflow:
+    name: PR Workflow
```
Comment (hoffa) [inline .github/workflows/build.yml]: nit: a more descriptive name would be helpful

Comment (hnnasit) [reply .github/workflows/build.yml]: I used `PR Workflow` to make it consistent with the other repos https://github.com/aws/aws-sam-cli/blob/develop/.github/workflows/build.yml#L11 and https://github.com/aws/aws-lambda-builders/blob/develop/.github/workflows/build.yml#L10. But I can change that to something like `Parent PR Status Check`.

Comment (hoffa) [reply .github/workflows/build.yml]: Consistency is great when it provides value; here I don't see what that value is. Your suggestion makes more sense.

Comment (hoffa) [reply .github/workflows/build.yml]: Also out of curiosity, why is this necessary? I would've thought the workflow fails anyway if any of the steps fail?

Comment (hnnasit) [reply .github/workflows/build.yml]: This is for easy maintenance of branch protection rules. Currently each check has to be manually added as a rule. But with a parent rule, new rules can just be added to the GH action and they don't need to be added in the branch protection rules manually.

**Review by hoffa — APPROVED** [review]

**Pull request merged**